File: Daily challenge/Boats to Save People.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) 
    {
        int count = 0;
        sort(people.begin(), people.end());
        int start =0;
        int end = people.size() - 1;
        for(start, end; start <= end; end--)
        {
            if(people[start] + people[end] <= limit)
            {
                start++;
            }
            count++;
        }
        return count;
    }
};

File: Daily challenge/Boats to Save People.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        count = 0
        people.sort()
        start, end = 0, len(people) - 1
        while start <= end:
            if people[start] + people[end] <= limit:
                start += 1
            count += 1
            end -= 1
        return count